Objective-C/CBLReplicator.mm

```diff
@@ -457,24 +457,24 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
             *error = [NSError errorWithDomain: CBLErrorDomain
                                          code: CBLErrorUnsupported
                                      userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
-        return [NSSet set];
+        return nil;
     }
     
     if (!_repl) {
         CBLLogInfo(Sync, @"Trying to fetch pending documentIds without a c4replicator %@", _repl);
-        return [NSSet set];
+        return nil;
     }
     
     C4Error c4err = {};
     C4SliceResult result = c4repl_getPendingDocIDs(_repl, &c4err);
     if (c4err.code > 0) {
         convertError(c4err, error);
         CBLWarnError(Sync, @"Error while fetching pending documentIds: %d/%d", c4err.domain, c4err.code);
-        return [NSSet set];
+        return nil;
     }
     
     if (result.size <= 0)
-        return [NSSet set];
+        return nil;
```

Comment:
If result's size == 0, it should return an empty set. This is not an error case.